src/v0/destinations/awin/utils.js

```diff
@@ -24,6 +26,51 @@ const getParams = (parameters, advertiserId) => {
   return params;
 };
 
+const areAllValuesDefined = (obj) =>
+  lodash.every(lodash.values(obj), (value) => !lodash.isUndefined(value));
+
+const buildProductPayloadString = (payload) => {
+  // URL-encode each value, and join back with the same key.
+  const encodedPayload = Object.entries(payload).reduce((acc, [key, value]) => {
+    // Encode each value. Assuming that all values are either strings or can be
+    // safely converted to strings.
+    acc[key] = encodeURIComponent(value);
+    return acc;
+  }, {});
+
+  return `AW:P|${encodedPayload.advertiserId}|${encodedPayload.orderReference}|${encodedPayload.productId}|${encodedPayload.productName}|${encodedPayload.productItemPrice}|${encodedPayload.productQuantity}|${encodedPayload.productSku}|${encodedPayload.commissionGroupCode}|${encodedPayload.productCategory}`;
+};
+
+// ref: https://wiki.awin.com/index.php/Advertiser_Tracking_Guide/Product_Level_Tracking#PLT_Via_Conversion_Pixel
+const trackProduct = (properties, advertiserId, commissionParts) => {
+  const transformedProductInfoObj = {};
+  if (properties?.products && properties.products.length > 0) {
```

Comment:
We should check if products is an array or not.

Reply:
done